src/archive/rar.rs

```diff
@@ -35,17 +47,22 @@ pub fn unpack_archive(archive_path: &Path, output_folder: &Path, quiet: bool) ->
 }
 
 /// List contents of `archive_path`, returning a vector of archive entries
-pub fn list_archive(archive_path: &Path) -> impl Iterator<Item = crate::Result<FileInArchive>> {
-    Archive::new(archive_path)
-        .open_for_listing()
-        .expect("cannot open archive")
-        .map(|item| {
-            let item = item?;
-            let is_dir = item.is_directory();
-            let path = item.filename;
-
-            Ok(FileInArchive { path, is_dir })
-        })
+pub fn list_archive(
+    archive_path: &Path,
+    password: Option<impl AsRef<[u8]>>,
+) -> impl Iterator<Item = crate::Result<FileInArchive>> {
+    let password = password.as_ref().map(|p| p.as_ref());
```

Comment:
I believe you can remove 4 places this mapping appears after replacing `Option<impl AsRef<[u8]>>` by `Option<&[u8]>`